docs/config-file/v2.rst

```diff
@@ -0,0 +1,365 @@
+
+.. code-block:: yaml
+
+   conda:
+     environment: environment.yml
+
+conda.environment
+`````````````````
+
+The path to the Conda environment file from the root of the project.
+
+:Type: ``path``
+:Required: ``true``
+
+build
+~~~~~
+
+Configuration for the documentation build process.
+
+.. code-block:: yaml
+
+   build:
+     image: latest
+
+   python:
+     version: 3.6
+
+
+build.image
+```````````
+
+The build Docker image to be used.
+
+:Type: ``string``
+:Options: ``stable``, ``latest``
+
+Each image support different Python versions and has different packages installed,
+as defined here:
+
+* `stable <https://github.com/rtfd/readthedocs-docker-images/tree/releases/2.x>`_: :supportedversions:`2.0`
+* `latest <https://github.com/rtfd/readthedocs-docker-images/tree/releases/3.x>`_: :supportedversions:`3.0`
+
+sphinx
+~~~~~~
+
+Configuration for Sphinx documentation
+(this is the default documentation type).
+
+.. code-block:: yaml
+
+   sphinx:
+     builder: html
+     configuration: conf.py
+     fail_on_warning: true
+
+sphinx.builder
+``````````````
+
+The builder type for the Sphinx documentation.
+
+:Type: ``string``
+:Options: ``html``, ``htmldir``, ``singlehtml``
+:Default: ``html``
+
+sphinx.configuration
+````````````````````
+
+The path to the ``conf.py`` file.
+
+:Type: ``path``
+:Default: ``null``
+
+If the value is ``null``,
+Read the Docs would try to find a ``conf.py`` file in your project.
+
+sphinx.fail_on_warning
+``````````````````````
+
+Turn warnings into errors.
+This means that the build stops at the first warning and exits with exit status 1.
+
+:Type: ``bool``
+:Default: ``false``
+
+mkdocs
+~~~~~~
+
+Configuration for Mkdocs documentation.
+
+.. code-block:: yaml
+
+   mkdocs:
+     configuration: mkdocs.yml
+     fail_on_warning: false
+
+mkdocs.configuration
+````````````````````
+
+The path to the ``mkdocs.yml`` file
+
+:Type: ``path``
+:Default: ``null``
+
+If the value is ``null``,
+Read the Docs would try to find a ``mkdocs.yml`` file in your project.
+
+mkdocs.fail_on_warning
+``````````````````````
+
+Turn warnings into errors.
+This means that the build stops at the first warning and exits with exit status 1.
+
+:Type: ``bool``
+:Default: ``false``
+
+submodules
+~~~~~~~~~~
+
+Git submodules configuration.
+
+.. note::
+   
+   You can't include and exclude submodules at the same time.
+
+.. code-block:: yaml
+   
+   submodules:
+     include: all
+     recursive: true
+
+submodules.include
+``````````````````
+
+List of submodules to be included.
+
+:Type: ``list``
+:Default: ``[]``
+
+.. note::
+   
+   You can use the ``all`` keyword to include all submodules.
+
+   .. code-block:: yaml
+
+      submodules:
+        include: all
+
+submodules.exclude
+``````````````````
+
+List of submodules to be excluded.
+
+:Type: ``list``
+:Default: ``[]``
+
+.. note::
+   
+   You can use the ``all`` keyword to exclude all submodules.
+   This is the same as ``include: []``.
```

Comment:
This line can be removed. It could be misinterpreted since it's not the "same" :)

Reply:
It is the same, both ways allow you to exclude all submodules

Reply:
Ohhhh! I didn't get it at first. You are right! :smile: